jenkins/promotion/release-promotion-parallel.jenkinsfile

```diff
@@ -0,0 +1,437 @@
+                    }
+                }
+            }
+        }
+        stage('Publish'){
+            parallel {
+                stage('OpenSearch Debian Apt promotion') {
+                    agent {
+                        docker {
+                            label AGENT_LINUX_X64
+                            image 'docker/library/alpine:3'
+                            registryUrl 'https://public.ecr.aws/'
+                            alwaysPull true
+                        }
+                    }
+                    steps {
+                        echo 'Triggering distribution-promote-repos for OpenSearch Debian Apt'
+                        build job: 'distribution-promote-repos', wait: true, parameters: [string(name: 'DISTRIBUTION_JOB_NAME', value: 'distribution-build-opensearch'), 
+                                                                                            string(name: 'DISTRIBUTION_REPO_TYPE', value: 'apt'),
+                                                                                            string(name: 'DISTRIBUTION_BUILD_NUMBER', value: params.OPENSEARCH_RC_BUILD_NUMBER),
+                                                                                            string(name: 'INPUT_MANIFEST', value: "${params.RELEASE_VERSION}/opensearch-${params.RELEASE_VERSION}.yml"),
+                                                                                        ]
+                        echo 'Promotion successful for Debian Apt OpenSearch!'
+                    }
+                }
+                stage('OpenSearch Yum promotion') {
+                    agent {
+                        docker {
+                            label AGENT_LINUX_X64
+                            image 'docker/library/alpine:3'
+                            registryUrl 'https://public.ecr.aws/'
+                            alwaysPull true
+                        }
+                    }
+                    steps {
+                        echo 'Triggering distribution-promote-repos for OpenSearch Yum'
+                        build job: 'distribution-promote-repos', wait: true, parameters: [string(name: 'DISTRIBUTION_JOB_NAME', value: 'distribution-build-opensearch'), 
+                                                                                            string(name: 'DISTRIBUTION_REPO_TYPE', value: 'yum'),
+                                                                                            string(name: 'DISTRIBUTION_BUILD_NUMBER', value: params.OPENSEARCH_RC_BUILD_NUMBER),
+                                                                                            string(name: 'INPUT_MANIFEST', value: "${params.RELEASE_VERSION}/opensearch-${params.RELEASE_VERSION}.yml"),
+                                                                                        ]
+                        echo 'Promotion successful for OpenSearch yum!'
+                    }
+                }
```

Comment:
Since we are repeating Steps for each stages with some variation, I am wondering if there is some better way to do this with some code restructuring. Such as defining stages and steps first and then iterating over them? 
I understand if that's totally not possible in scripts as this.

Reply:
I believe it is possible by putting it into groovy scripts but we want to control each job execution and parameters at the lower level here. Definitely there is a scope for improvement once we are sure of the process.

Reply:
nice, maybe worth exploring this separately?